crates/miden-objects/src/note/note_tag.rs

```diff
@@ -180,41 +218,42 @@ impl NoteTag {
             return Err(NoteError::NoteTagUseCaseTooLarge(use_case_id));
         }
 
-        let execution_bits = LOCAL_EXECUTION_WITH_ALL_NOTE_TYPES_ALLOWED;
         let use_case_bits = (use_case_id as u32) << 16;
         let payload_bits = payload as u32;
 
-        Ok(Self(execution_bits | use_case_bits | payload_bits))
+        Ok(Self::LocalAny(LOCAL_ANY | use_case_bits | payload_bits))
     }
 
     // PUBLIC ACCESSORS
     // --------------------------------------------------------------------------------------------
 
-    /// Returns true if the note is intended for execution by a specific account.
-    ///
-    /// A note is intended for execution by a single account if the first two bits are zeros
+    /// Returns true if the note is intended for execution by a specific account, i.e.
+    /// [`NoteTag::NetworkAccount`]
     pub fn is_single_target(&self) -> bool {
-        let first_2_bit = self.0 >> 30;
-        first_2_bit == 0b00
+        matches!(self, NoteTag::NetworkAccount(_))
     }
 
     /// Returns note execution mode defined by this tag.
     ///
     /// If the most significant bit of the tag is 0 the note is intended for local execution;
     /// otherwise, the note is intended for network execution.
     pub fn execution_mode(&self) -> NoteExecutionMode {
-        let first_bit = self.0 >> 31;
-
-        if first_bit == (LOCAL_EXECUTION as u32) {
-            NoteExecutionMode::Local
-        } else {
-            NoteExecutionMode::Network
+        match self {
+            NoteTag::NetworkAccount(_) | NoteTag::NetworkUseCase(..) => NoteExecutionMode::Network,
+            NoteTag::LocalAny(_) | NoteTag::LocalPublicAny(..) => NoteExecutionMode::Local,
         }
     }
 
     /// Returns the inner u32 value of this tag.
-    pub fn inner(&self) -> u32 {
-        self.0
+    pub fn as_u32(&self) -> u32 {
+        match self {
```

Comment:
```suggestion
    pub fn as_u32(&self) -> u32 {
        // Note that we always set the two most significant bits to the prefix corresponding to the
        // enum variant. See the note on type safety on the NoteTag docs.
        match self {
```

Let's add a short explanation.